### PR TITLE
editoast: move rolling stock and rs with liveries to schemas to schemas

### PR DESCRIPTION
**Pull request by Wadjetz**
part of #7048

### CODE REVIEW

File: editoast/editoast_schemas/src/rolling_stock/mod.rs [hunk 1]
```diff
@@ -34,7 +34,28 @@ mod rolling_stock_livery;
...
+    pub id: i64,
+    #[serde(flatten)]
+    pub common: RollingStockCommon,
+    pub railjson_version: String,
+    /// Whether the rolling stock can be edited/deleted or not.
+    pub locked: bool,
+    pub metadata: RollingStockMetadata,
+}
+
 editoast_common::schemas! {
```
Comment (leovalais) [inline editoast/editoast_schemas/src/rolling_stock/mod.rs]: I'd prefer to keep that declaration just below the imports.

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks for this PR. There are a few things to change, including some changes that went while I was off:

1. As stated in the tracking issue https://github.com/osrd-project/osrd/issues/7048#issue-2213585803, there should be only one definition of RollingStock in editoast_schemas.
2. RollingStockWithLiveries is a view response, not a schema. It should be moved into views/.
3. Please reference the tracking issues in the description of the PR, that helps the review ;)

**Review by woshilapin — APPROVED** [review]
Looks good. Thanks.

File: editoast/src/views/rolling_stocks/mod.rs [hunk 2]
```diff
@@ -633,9 +641,30 @@ pub mod tests {
...
+            .mass(response_body.mass)
+            .rolling_resistance(response_body.rolling_resistance)
+            .loading_gauge(response_body.loading_gauge)
+            .power_restrictions(response_body.power_restrictions)
+            .energy_sources(response_body.energy_sources)
+            .electrical_power_startup_time(response_body.electrical_power_startup_time)
+            .raise_pantograph_time(response_body.raise_pantograph_time)
+            .supported_signaling_systems(response_body.supported_signaling_systems);
+
+        assert_eq!(response_body.name, Some(rolling_stock_form.name));
```
Comment (leovalais) [inline editoast/src/views/rolling_stocks/mod.rs]: The changeset conversion doesn't seem necessary.

File: editoast/editoast_schemas/src/rolling_stock/mod.rs
Comment (leovalais) [inline editoast/editoast_schemas/src/rolling_stock/mod.rs:1]: As the split progresses we want to adopt the Rust 2018 modules convention. Could you mv this file to `editoast_schemas/src/rolling_stock.rs`?

File: editoast/editoast_schemas/src/rolling_stock/mod.rs (hunk 1 above)
Comment (leovalais) [reply editoast/editoast_schemas/src/rolling_stock/mod.rs]: I'd also prefer to have a `schema!` declaration in each module a schema is defined. 

```rust
schema! {
    RollingStock,
    gamma:schemas(),
    energy_source::schemas()
    ...
}
```

**Review by leovalais — CHANGES_REQUESTED** [review]

File: editoast/editoast_schemas/src/rolling_stock/mod.rs
Comment (Wadjetz) [reply editoast/editoast_schemas/src/rolling_stock/mod.rs:1]: yes, but it trigger clippy modules inception warning

File: editoast/src/views/rolling_stocks/mod.rs [hunk 3]
```diff
@@ -633,9 +640,8 @@ pub mod tests {
         // Check rolling_stock creation
         let response_body: RollingStock = assert_status_and_read!(post_response, StatusCode::OK);
         let rolling_stock_id: i64 = response_body.id;
-        let response_body: Changeset<RollingStockModel> = response_body.into();
 
-        assert_eq!(response_body.name, Some(rolling_stock_form.common.name));
+        assert_eq!(Some(response_body.name), Some(rolling_stock_form.name));
```
Comment (leovalais) [inline editoast/src/views/rolling_stocks/mod.rs]:
```suggestion
        assert_eq!(response_body.name, rolling_stock_form.name);
```

**Review by leovalais — APPROVED** [review]
LGTM thanks!

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Pull request merged**